R/train_and_evaluate.R

```diff
@@ -38,19 +38,46 @@
 #' * ValueError: if environment variable `TF_CONFIG` is incorrectly set.
 #' 
 #' @family training methods
-#' 
+#' @export
 train_and_evaluate.tf_estimator <- function(object, train_spec, eval_spec) {
   if (tf_version() < '1.4') {
     stop("train_and_evaluate() is only available since TensorFlow v1.4")
   }
   estimator <- object$estimator
   train_spec$args$input_fn <- normalize_input_fn(object, train_spec$args$input_fn)
   eval_spec$args$input_fn <- normalize_input_fn(object, eval_spec$args$input_fn)
-  tf$estimator$train_and_evaluate(
-    estimator = estimator,
-    train_spec = do.call(tf$estimator$TrainSpec, train_spec$args),
-    eval_spec = do.call(tf$estimator$EvalSpec, eval_spec$args)
-  )
+  with_logging_verbosity(tf$logging$WARN, {
+    tf$estimator$train_and_evaluate(
+      estimator = estimator,
+      train_spec = do.call(tf$estimator$TrainSpec, train_spec$args),
+      eval_spec = do.call(tf$estimator$EvalSpec, eval_spec$args)
+    )
+  })
+  
+  training_history <- as.data.frame(.globals$history[[mode_keys()$TRAIN]])
+  steps_per_axis_unit <- training_history$steps[2] - training_history$steps[1]
+  
+  training_history <- if (nrow(training_history) > 100) {
+    # cap number of points plotted
```

Comment:
@jjallaire FYI here's what motivated my question regarding time axis labels. The current solution is to cap the number of rows being written to the training history (and hence the number of points plotted in `tfruns`). I'm also leaving out the `steps` column so it doesn't get plotted, and writing a `steps_per_axis_unit` so users can interpret the plot (may need a new section in the `view_run()` dashboard to show this). Ideally we'd have actual step units on the time axis without blowing up the browser.

Reply:
Should be addressed by https://github.com/rstudio/tfruns/commit/1d25ab63e8cfb196b30b3e5af0656e21f961d145